qemu/tests/usb_host.py

```diff
@@ -67,20 +67,24 @@ def usb_dev_unplug():
                           " returns: '%s'" % reply)
         return
 
-    device = params["usb_host_device"]
-    (vendorid, productid) = device.split(":")
+    usb_hostdev = params["usb_devices"].split()[-1]
+    usb_options = params.get("options")
+    if usb_options == "with_vendorid_productid":
+        vendorid = params["vendorid_%s" % usb_hostdev]
+        productid = params["productid_%s" % usb_hostdev]
+        usb_options = "vendorid=%s,productid=%s" % (vendorid, productid)
 
     # compose strings
-    lsusb_cmd = "lsusb -v -d %s" % device
-    monitor_add = "device_add usb-host,bus=usbtest.0,id=usbhostdev"
+    lsusb_cmd = "lsusb -v -d %s:%s" % (vendorid, productid)
```

Comment:
It can only work with `usb_options == "with_vendorid_productid"`.

Reply:
Actually no. In 'with_hostbus_hostaddr' case, we convert the hostbus/hostaddr to vendorid/productid.

Reply:
> with_hostbus_hostaddr

I am mainly talking about `with_no_hostdev` not other combination.
This patch include 5 commits and this code was added in the first commit.  I am checking the commit one by one, we need ensure every one can work not only for the final change.

 hostbus and hostport combination still have not be added in the first commit.

Reply:
In the first commit, only case `usb.usb_host..usb_negative_test`  can fall into `with_no_hostdev`, lead to `if params.get("usb_negative_test", "no") != "no":` code block.
It will return in the block, so have no chance to go to this line `lsusb_cmd = "lsusb -v -d %s:%s" % (vendorid, productid)`.

Reply:
Got it, thanks!